src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/BeneficialOwnerChangeService.java

```diff
@@ -0,0 +1,435 @@
+    }
+
+    ChangeManager<CorporateBeneficialOwnerPsc, PscApi, PrivateBoDataApi> changeManager = new ChangeManager<>(
+        psc, publicPrivateBoPair);
+
+    var beneficialOwnerNatureOfControlTypes = beneficialOwnerCorporateDto.getBeneficialOwnerNatureOfControlTypes();
+    var trusteesNatureOfControlTypes = beneficialOwnerCorporateDto.getTrusteesNatureOfControlTypes();
+    var nonLegalFirmMembersNatureOfControlTypes = beneficialOwnerCorporateDto.getNonLegalFirmMembersNatureOfControlTypes();
+
+    var collectedNatureOfControl = NatureOfControlTypeMapping.collectAllNatureOfControlsIntoSingleList(
+        beneficialOwnerNatureOfControlTypes,
+        trusteesNatureOfControlTypes, nonLegalFirmMembersNatureOfControlTypes);
+
+    boolean hasChange = setCommonAttributes(changeManager,
+        publicPrivateBoPair,
+        beneficialOwnerCorporateDto.getServiceAddress(),
+        beneficialOwnerCorporateDto.getPrincipalAddress(),
+        collectedNatureOfControl);
+
+    hasChange |= changeManager.compareAndBuildLeftChange(
+        beneficialOwnerCorporateDto.getName(),
+        PscApi::getName,
+        CorporateBeneficialOwnerPsc::setCorporateName);
+
+    Identification publicIdentification = publicPrivateBoPair.getLeft() == null ? null
+        : publicPrivateBoPair.getLeft().getIdentification();
+
+    ChangeManager<CompanyIdentification, PscApi, Identification> companyIdentificationChangeManager
+        = new ChangeManager<>(psc.getCompanyIdentification(),
+        new ImmutablePair<>(publicPrivateBoPair.getLeft(), publicIdentification));
+
+    hasChange |= companyIdentificationChangeManager.compareAndBuildPrivateChange(
+        beneficialOwnerCorporateDto.getLegalForm(),
+        Identification::getLegalForm,
+        CompanyIdentification::setLegalForm);
+
+    hasChange |= companyIdentificationChangeManager.compareAndBuildPrivateChange(
+        beneficialOwnerCorporateDto.getLegalForm(),
+        Identification::getLegalForm,
+        CompanyIdentification::setLegalForm);
+
+    hasChange |= companyIdentificationChangeManager.compareAndBuildPrivateChange(
+        beneficialOwnerCorporateDto.getLawGoverned(),
+        Identification::getLegalAuthority,
+        CompanyIdentification::setGoverningLaw);
+
+    hasChange |= companyIdentificationChangeManager.compareAndBuildPrivateChange(
+        beneficialOwnerCorporateDto.getPublicRegisterName(),
+        Identification::getPlaceRegistered,
+        CompanyIdentification::setRegisterLocation);
+
+    hasChange |= companyIdentificationChangeManager.compareAndBuildPrivateChange(
+        beneficialOwnerCorporateDto.getRegistrationNumber(),
+        Identification::getRegistrationNumber,
+        CompanyIdentification::setRegistrationNumber);
+
+    hasChange |= companyIdentificationChangeManager.compareAndBuildPrivateChange(
+        beneficialOwnerCorporateDto.getPublicRegisterName(),
+        Identification::getPlaceRegistered,
+        CompanyIdentification::setPlaceRegistered);
```

Comment:
This block of code looks duplicated from 211, but the last line is different? (setPlaceRegistered)

Reply:
211 is getRegistration**Number** this is getPublicRegister**Name**